src/main/java/org/spdx/sbom/gradle/SpdxSbomPlugin.java

```diff
@@ -129,13 +128,29 @@ private void createTaskForTarget(
                   t.getSpdxKnownLicensesService().set(knownLicenseServiceProvider);
                   t.usesService(knownLicenseServiceProvider);
 
+                  final String artifactType;
+                  if (project.getPlugins().hasPlugin("com.android.base")) {
+                    artifactType = "android-aar-or-jar";
+                  } else {
+                    artifactType = "jar";
+                  }
+
                   List<String> configurationNames = target.getConfigurations().get();
+                  var rootComponentsProperty =
+                      project.getObjects().listProperty(ResolvedComponentResult.class);
                   for (var configurationName : configurationNames) {
                     Provider<Set<ResolvedArtifactResult>> artifacts =
                         project
                             .getConfigurations()
                             .getByName(configurationName)
                             .getIncoming()
+                            .artifactView(
+                                viewConfiguration ->
+                                    viewConfiguration.attributes(
+                                        attributes ->
+                                            attributes.attribute(
+                                                Attribute.of("artifactType", String.class),
+                                                artifactType)))
```

Comment:
Maybe it's possible to map the POM info into this and get a unique set of the `packaging` from the POM and use that to set up the attributes. Not sure if that'll work, but it's an idea :sweat_smile:

Reply:
Can we just apply this filter in the android usecase? It appears this only happens when referencing a subproject from within an android project -- because android subprojects output all these crazy variants?

```
if android project {
  apply artifactView
}
else {
  just do what we were doing before
}
```

Reply:
I'm just not super fond of applying the filter and potentially missing some cases (war, zip, rar, etc)

Reply:
I think just applying the filter in the Android use case would work for now until there's another type of plugin that causes the same issue.

I'm experimenting with making filters based on the packaging in the poms.

Reply:
I was able to get it working by using a separate artifactView for each packaging found in the POMs, but it required some provider magic redirections (i.e. declare a property for the set of packagings at the top, flatMapping it to the artifactViews, and populating the packagings property later on in the file mapping from the pomInfo property).

It seems to be working pretty well. I tested on a regular Kotlin JVM project, a Kotlin JVM project that depends on a KMP project, and a KMP project, and it all worked fine. For the KMP project, using the jvm configuration resulted in an sbom referencing jar files, and using the js configuration resulted in an sbom referencing klib files (which was pretty cool).

Android was a trouble maker again, but that was because there are conflicts where requesting jar artifacts returns the classes.jar from the aar file. The workaround was to detect if there was an Android plugin and replace aar and jar with android-aar-or-jar in the list of packagings.

It seems to be working with the configuration cache, and from what I can tell the resolution is happening during the configuration phase, so it should be fine. Let me know what you want to do with that.